src/shared/components/Graph/Graph.ts

```diff
@@ -0,0 +1,39 @@
+import graph from './d3-graph';
+/**
+ * This class will handle stateful aspects of each d3 class instance
+ * I did this to make sure React can keep track of instances and destroy them
+ * as well as report errors
+ *
+ * @export
+ * @class Graph
+ */
+export class Graph {
+  private renderError?: Error;
+  readonly dotGraph: string;
+  readonly element: HTMLElement;
+  private destroy?: () => void;
+  constructor(dotGraph: string, element: HTMLElement) {
+    this.dotGraph = dotGraph;
+    this.element = element;
+  }
+
+  render() {
+    try {
+      this.destroy = graph(this.dotGraph, this.element);
+    } catch (error) {
+      // tslint:disable-next-line:no-console
```

Comment:
Be careful with this, it might end up in the code base. You can skip the githook when you push instead. That way we can insure no dead code makes it through

Reply:
That's cool, but this particular error I want to log, in case we get some interesting data that breaks the parser.

Reply:
Ok, not in prod, we'll use another consolidated logging service somehow 👍